src/main/java/io/quarkus/gizmo2/impl/BlockCreatorImpl.java

```diff
@@ -269,20 +270,17 @@ public void remAssign(final LValueExpr var, final Expr arg) {
         set(var, rem(var, arg));
     }
 
-    private ClassDesc boxType(TypeKind typeKind) {
-        return switch (typeKind) {
-            case BOOLEAN -> CD_Boolean;
-            case BYTE -> CD_Byte;
-            case CHAR -> CD_Character;
-            case SHORT -> CD_Short;
-            case INT -> CD_Integer;
-            case LONG -> CD_Long;
-            case FLOAT -> CD_Float;
-            case DOUBLE -> CD_Double;
-            case VOID -> CD_Void;
-            default -> throw new IllegalArgumentException("No box type for " + typeKind);
-        };
-    }
+    private static final Map<ClassDesc, ClassDesc> boxTypes = Map.of(
+        CD_boolean, CD_Boolean,
+        CD_byte, CD_Byte,
+        CD_char, CD_Character,
+        CD_short, CD_Short,
+        CD_int, CD_Integer,
+        CD_long, CD_Long,
+        CD_float, CD_Float,
+        CD_double, CD_Double,
+        CD_void, CD_Void
```

Comment:
If you can't box/unbox `void` then why this map entry?

Reply:
That is a great question that I have asked myself as well, and frankly, I'm not sure. It's been there, so I kept it :-) I guess there might eventually be a usecase that would require it, but that's a great case of YAGNI...

Reply:
We would use it when autoboxing return types, probably. I think it was being used at one point for something else, maybe? But you can remove it if you want; we can always add it back if needed.

I would suggest that the second map be built from the first map though, rather than duplicating the map entries.

Reply:
Makes sense, will do.

Reply:
Done. I replaced the second map by a call to `reverseMap()`. I've kept the `void`/`Void` types in the map just to avoid possible future problems when we have to add it. It's there since the beginning, so everyone has to deal with it.